src/Language/Haskell/Exts/FreeVars.hs

```diff
@@ -28,9 +30,14 @@ import           Prelude
 
 data Vars = Vars {bound :: Set (Name ()), free :: Set (Name ())}
 
+instance Semigroup Vars where
+    Vars x1 x2 <> Vars y1 y2 = Vars (x1 ^+ y1) (x2 ^+ y2)
+
 instance Monoid Vars where
     mempty = Vars Set.empty Set.empty
-    mappend (Vars x1 x2) (Vars y1 y2) = Vars (x1 ^+ y1) (x2 ^+ y2)
+#if !(MIN_VERSION_base(4,11,0))
+    mappend = (<>)
```

Comment:
I thought this definition of mappend was fine for all compilers? And thus doesn't need CPP.

Reply:
True. I'll simplify and merge

Reply:
Sorry, I left out that detail. Not only is `Semigroup` a superclass of `Monoid`, but the default definition of `mappend` is now `(<>)`, so defining `mappend` explicitly is redundant. (In a future version of GHC, `mappend` may even be removed as a class method of `Monoid` entirely, so this saves us some effort down the line.)